packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/web_resource_response.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// Contains information about the response for a request.
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// This example demonstrates how to extend the [WebResourceResponse] to
+/// provide additional platform specific parameters.
+///
+/// When extending [WebResourceResponse] additional parameters should always
+/// accept `null` or have a default value to prevent breaking changes.
+///
+/// ```dart
+/// class AndroidWebResourceResponse extends WebResourceResponse {
+///   WebResourceResponse._({
+///     required WebResourceResponse response,
+///   }) : super(
+///     uri: response.uri,
+///     statusCode: response.statusCode,
+///     headers: response.headers,
+///   );
+///
+///   factory AndroidWebResourceResponse.fromWebResourceResponse(
+///     WebResourceResponse response, {
+///     Uri? historyUrl,
+///   }) {
+///     return AndroidWebResourceResponse._(response, historyUrl: historyUrl);
+///   }
+///
+///   final Uri? historyUrl;
+/// }
+/// ```
+@immutable
+class WebResourceResponse {
+  /// Used by the platform implementation to create a new [WebResourceResponse].
+  const WebResourceResponse({
+    required this.uri,
+    required this.statusCode,
+    this.headers = const <String, String>{},
+  });
+
+  /// The URI that this response is associated with.
+  final String? uri;
```

Comment:
Sorry, I know I said we should go back to `String?` above, but now that I'm looking at it again in the context of not asking it to be changed to a request object: shouldn't it be `Uri?`?